src/lib/pubkey/classic_mceliece/cmce_gf.cpp

```diff
@@ -0,0 +1,84 @@
+/*
+* Classic McEliece GF arithmetic
+* Based on the public domain reference implementation by the designers
+* (https://classic.mceliece.org/impl.html - released in Oct 2022 for NISTPQC-R4)
+*
+* (C) 2023 Jack Lloyd
+*     2023,2024 Fabian Albert, Amos Treiber - Rohde & Schwarz Cybersecurity
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+**/
+
+#include <botan/internal/cmce_gf.h>
+
+namespace Botan {
+
+namespace {
+// Only for moduli 0b0010000000011011 and 0b0001000000001001
+inline CmceGfElem internal_reduce(uint32_t x, CmceGfMod mod) {
```

Comment:
Would it make sense for `mod` to be a compile-time template parameter instead?

Alternately, if `CmceGfMod` could only take on these specific values we could be assured that unexpected inputs could not occur.